contracts/lbp/LBPManager.sol

```diff
@@ -0,0 +1,290 @@
+                true // SwapEnabled is set to true at pool creation.
+            )
+        );
+
+        lbp.updateWeightsGradually(
+            _startTimeEndTime[0],
+            _startTimeEndTime[1],
+            _endWeights
+        );
+
+        return address(lbp);
+    }
+
+    /**
+     * @dev                             Subtracts the primeDaoFee and adds liquidity to the LBP.
+     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _projectTokenIndex        Index for the _tokenList array for the funding token.
+     * @param _sender                   Address of the liquidity provider.
+     * @param _userData                 UserData object that specifies the type of join.
+     */
+    function addLiquidity(
+        IERC20[] memory _tokenList,
+        uint8 _projectTokenIndex,
+        address _sender,
+        bytes memory _userData
+    ) external onlyAdmin {
+        require(!poolFunded, "LBPManager: pool has already been funded");
+        poolFunded = true;
+
+        IVault vault = lbp.getVault();
+
+        if (primeDaoFeePercentage != 0) {
+            // Transfer primeDaoFee to beneficiary.
+            _tokenList[_projectTokenIndex].transferFrom(
+                _sender,
+                beneficiary,
+                _feeAmountRequired(_projectTokenIndex)
+            );
+        }
+
+        for (uint8 i; i < _tokenList.length; i++) {
+            _tokenList[i].transferFrom(_sender, address(this), amounts[i]);
+            _tokenList[i].approve(address(vault), amounts[i]);
+        }
+
+        IVault.JoinPoolRequest memory request = IVault.JoinPoolRequest({
+            maxAmountsIn: amounts,
+            userData: _userData,
+            fromInternalBalance: false, // It is not possible to add liquidity through the internal Vault balance.
+            assets: _tokenList
+        });
+
+        vault.joinPool(lbp.getPoolId(), address(this), address(this), request);
+    }
+
+    /**
+     * @dev                             exit pool or remove liquidity from pool
+     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _receiver                 Address of the liquidity receiver, after exiting the LBP.
+     * @param _toInternalBalance        Balancer's Vault option to send the liquidity from the LBP to the _receivers internal Vault balance.
+     * @param _userData                 UserData object that specifies the type of exit.
+     */
+    function removeLiquidity(
```

Comment:
If we add a new function, it should be always tested in the same PR, as I understand there are no tests rn?

Reply:
Yep, there are, missed tests(test file wasn't displayed because of +1000 loc)